### PR TITLE
OHRM5X-2056: Develop My Claims screen

### PR DESCRIPTION
**Pull request by DulsaraNethmin**

### CODE REVIEW

File: installer/Migration/V5_5_0/permission/screens.yaml [hunk 1]
```diff
@@ -3,21 +3,32 @@
   url: viewClaimModule
   permissions:
     - { role: Admin, permission: { read: true } }
-    - { role: ESS, permission: { read: true, create: false, update: false, delete: false }}
+    - {
+        role: ESS,
+        permission: { read: true, create: false, update: false, delete: false },
+      }
```
Comment (ManulMax) [inline installer/Migration/V5_5_0/permission/screens.yaml:9]: Not sure why this changing everytime

File: src/client/src/orangehrmClaimPlugin/components/AddAttachmentModal.vue [hunk 2]
```diff
@@ -22,7 +22,7 @@
   <oxd-dialog class="orangehrm-dialog-modal" @update:show="onCancel">
     <div class="orangehrm-modal-header">
       <oxd-text type="card-title">
-        {{ $t('general.add_attachment') }}
+        {{ $t('claim.add_attachment') }}
```
Comment (ManulMax) [inline src/client/src/orangehrmClaimPlugin/components/AddAttachmentModal.vue:25]: installer/Migration/V5_0_0/lang-string/general.yaml

**Review by ManulMax — CHANGES_REQUESTED** [review]

**Review by ManulMax — CHANGES_REQUESTED** [review]

File: src/client/src/orangehrmClaimPlugin/components/StatusDropdown.vue [hunk 3]
```diff
@@ -38,19 +38,23 @@ export default {
...
       },
       {
         id: 'REJECTED',
-        label: this.$t('claim.rejected'),
+        label: this.$t('leave.rejected'),
       },
       {
         id: 'CANCELLED',
-        label: this.$t('claim.cancelled'),
+        label: this.$t('leave.cancelled'),
```
Comment (RajithaKumara) [inline src/client/src/orangehrmClaimPlugin/components/StatusDropdown.vue]: seems we have to move these labels to `general` as it's using multiple places

File: src/client/src/orangehrmClaimPlugin/components/ClaimExpenses.vue [hunk 4]
```diff
@@ -226,9 +226,7 @@ export default {
       return computedHeaders;
     },
     formatedAmount() {
-      let value = this.totalAmount.replace(/,/g, '');
-      value = value.replace(/\B(?=(\d{3})+(?!\d))/g, ',');
-      return value;
+      return this.totalAmount.replace(/\B(?=(\d{3})+(?!\d))/g, ',');
```
Comment (RajithaKumara) [inline src/client/src/orangehrmClaimPlugin/components/ClaimExpenses.vue:229]: check this,
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString

File: src/client/src/core/util/validation/__tests__/rules.spec.ts [hunk 5]
```diff
@@ -1073,7 +1073,7 @@ describe('core/util/validation/rules::digitsOnlyWithTwoDecimalPoints', () => {
 
   test('digitsOnlyWithTwoDecimalPoints:: with one decimal points', () => {
     const result = digitsOnlyWithTwoDecimalPoints('123.9');
```
Comment (RajithaKumara) [inline src/client/src/core/util/validation/__tests__/rules.spec.ts:1075]: is this converted to `123.90` when saving or save `123.9` as it is

File: src/client/src/orangehrmClaimPlugin/components/AddExpenseModal.vue [hunk 6]
```diff
@@ -170,7 +170,7 @@ export default {
         .create({
           expenseTypeId: this.expense.type.id,
           date: this.expense.date,
-          amount: (Math.floor(this.expense.amount * 100) / 100).toFixed(2),
+          amount: this.expense.amount,
```
Comment (RajithaKumara) [inline src/client/src/orangehrmClaimPlugin/components/AddExpenseModal.vue:173]:
```suggestion
          amount: this.expense.amount.toFixed(2),
```
:thinking: I'm not sure whether we need to add this

File: src/client/src/orangehrmClaimPlugin/components/EditExpenseModal.vue [hunk 7]
```diff
@@ -186,7 +186,7 @@ export default {
         .update(this.data.id, {
           expenseTypeId: this.selectedOption.id,
           date: this.expense.date,
-          amount: (Math.floor(this.expense.amount * 100) / 100).toFixed(2),
+          amount: this.expense.amount,
```
Comment (RajithaKumara) [inline src/client/src/orangehrmClaimPlugin/components/EditExpenseModal.vue:189]: *

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/client/src/core/util/validation/__tests__/rules.spec.ts (hunk 5 above)
Comment (DulsaraNethmin) [reply src/client/src/core/util/validation/__tests__/rules.spec.ts:1075]: yes

File: src/client/src/orangehrmClaimPlugin/components/AddExpenseModal.vue (hunk 6 above)
Comment (Super-Chama) [reply src/client/src/orangehrmClaimPlugin/components/AddExpenseModal.vue:173]: @RajithaKumara is correct, if number is coming in expense.amount, no need to do these additional calculations. just use `toFixed()`

image

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**